client.js

```diff
@@ -170,22 +171,27 @@ class ChordCrawler {
     this.walk.clear();
   }
   async crawl() {
-    const connectionString = `${this.host}:${this.port}`;
-    console.log(`Connecting to ${connectionString}`);
-    const client = caller(connectionString, PROTO_PATH, "Node");
-
-    try {
-      const successorNode = await client.getSuccessorRemoteHelper(
-        DUMMY_REQUEST_OBJECT
-      );
-      this.pruneUponCycle(connectionString);
-      this.updateSuccessor(connectionString, successorNode);
-      this.updateNode(successorNode);
-      this.host = successorNode.host;
-      this.port = successorNode.port;
-    } catch (err) {
-      console.error("Error is : ", err);
-      this.shuffleCurrentNode();
+    if (this.canAdvance) {
+      this.canAdvance = false;
+      const connectionString = `${this.host}:${this.port}`;
+      console.log(`Connecting to ${connectionString}`);
+      const client = caller(connectionString, PROTO_PATH, "Node");
+
+      try {
+        const successorNode = await client.getSuccessorRemoteHelper(
```

Comment:
info: just as a general comment, you could just call "getSuccessor(<crawler object>, <remote object>)" and it will do this for you.  the idea was to have a "single interface" and let the wrapper (e.g., getSuccessor()) deal with the intricacies.  but that would break your current concept of the "connectionString" since the objects are nodes of the form {id, host, port}.  again, just FYI.